### PR TITLE
runtime_vm: Apply the correct label before the sandbox is created

### PR DESCRIPTION
**Pull request by fidencio**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

Although the current code does apply the correct label, it does it too
late, after the Sandbox has been created and the processes already being
ran.

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

Testing this PR is not something easy without a previously setup environment.
As I have one handy, please, let me know via a private message and I can give you access to it.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by fidencio** [comment]
@rhatdan, @haircommander, please, take a look.

**Comment by haircommander** [comment]
can you fix the issues with
`make lint`?
also, is there any harm in doing both? wouldn't a new container process running in the rm also need the correct labels?

**Comment by fidencio** [comment]
> can you fix the issues with
> `make lint`?

Sure!

> also, is there any harm in doing both? wouldn't a new container process running in the vm also need the correct labels?

There's no harm, @haircommander, but it's not useful either. The reason for that is kata not applying any SELinux labels on the VM workloads: https://github.com/kata-containers/runtime/blob/master/virtcontainers/kata_agent.go#L1102

What's your preference?

**Comment by haircommander** [comment]
ah I see, can you add a comment making note of that?

**Comment by fidencio** [comment]
> ah I see, can you add a comment making note of that?

Sure!

**Comment by haircommander** [comment]
nvm on the lint, the only lint issue is fixed in https://github.com/cri-o/cri-o/pull/3887

**Comment by haircommander** [comment]
integration failures are not the fault of this PR

File: server/sandbox_run_linux.go [hunk 1]
```diff
@@ -482,6 +484,23 @@ func (s *Server) runPodSandbox(ctx context.Context, req *pb.RunPodSandboxRequest
 	if err != nil {
 		return nil, err
 	}
+
+	runtimeType, err := s.Runtime().ContainerRuntimeType(container)
+	if err != nil {
+		return nil, err
+	}
+	// If using kata runtime, the process label should be set to container_kvm_t
+	// Mind that kata does *not* apply any process label to the VM workloads.
```
Comment (haircommander) [inline server/sandbox_run_linux.go:493]: suggested rewording:
`Keep in mind that kata does *not* apply any process label to containers within the VM`

Comment (fidencio) [reply server/sandbox_run_linux.go]: Suggestion taken!

**Comment by haircommander** [comment]
test failures are not the fault of this PR, but are legit. they're fixed by https://github.com/opencontainers/runc/pull/2479

**Comment by fidencio** [comment]
I've rebased this one atop of the origin/main and forced pushed, in order to see if it improves the issues with the CI which are not related to the PR itself.

**Pull request merged**